src/main/MainTabsScreen.js

```diff
@@ -153,4 +163,4 @@ function withHaveServerDataGate(Comp) {
   );
 }
 
-export default withHaveServerDataGate(MainTabsScreen);
+export default withHaveServerDataGate<Props, ComponentType<Props>>(MainTabsScreen);
```

Comment:
I think you can simplify this like so:
```js
export default withHaveServerDataGate<Props, _>(MainTabsScreen);
```
and let it infer the second type argument as `ComponentType<Props>`.

We'll basically always want it to be that, which is also the totally boring thing for it to be. So I think a convention of saying just `_` is the most helpful one -- like we do with `connect`.

Oh, in fact, we can simplify further: drop `C: ComponentType<$Exact<P>>` from the definition, and just substitute `ComponentType<$Exact<P>>` in the one place the definition mentions `C`.

In the type of `connect` I do something like this `C` argument, because it appears several times and there's a readability benefit to not repeating it. But it has this trade-off that it awkwardly adds a type parameter to the interface. Here we can just skip that.